hydrolib/core/io/ini/util.py

```diff
@@ -462,3 +462,28 @@ def is_valid_coordinates_with_num_coordinates_specification() -> bool:
         raise ValueError(error)
 
     return root_validator(allow_reuse=True)(validate_location_specification)
+
+
+def get_key_renaming_root_validator(keys_to_rename: Dict[str, List[str]]):
```

Comment:
I think this functionality can also be used for non-ini files. 
We could move this to a different namespace in another issue, since it is not important now. 

Also, I like the current generic approach.

I think in the future we could introduce a dedicated class of which an instance can be passed to this function.
We should also try to make use the semantic versioning from the files to do backwards compatibility, this will make it more reliable over the lifespan of hydrolib-core.

Reply:
Agreed. I talked about this validator being useful for non-ini files with Arthur. We decided we can make it more generic when we need it. So for now we wanted to just have it for the ini files.

Also completely agree with your second point. It would cleaner if we pass an instance of a (new, to be created) class to this function.

One of the problems of the versioning of the files is that they are not consistant. Take a look, for example, at the documentation in the 1D2D and 2D3D manuals and look for the .bc file documentation. Both have fileversion 1.01, but they use different keywords :(

Shall I make a new issue for improving the this validator? Or do we just create a new issue later, when/if we decide to work on it?

Reply:
We will create a follow-up issue in which we will think about our requirements for backwards compatibility. After that, we can improve this implementation. For now: no action needed.